### PR TITLE
(CDAp-15787) Alert plugin

### PR DESCRIPTION
**Pull request by MEseifan**
Updated Alert plugin to use FailureCollector

### CODE REVIEW

File: core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java [hunk 1]
```diff
@@ -136,13 +144,15 @@ private Conf() {
       maxAlertsPerSecond = 100;
     }
 
-    private void validate() {
+    private void validate(FailureCollector collector) {
       if (autoCreateTopic && namespace != null) {
-        throw new IllegalArgumentException("Cannot auto-create topic when namespace is set.");
+        collector.addFailure("Cannot auto-create topic when namespace is set.",
+                "Namespace must not be set to auto-create topic.").withConfigProperty("namespace")
```
Comment (albertshau) [inline core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java]: style: fix indentation

```
collector.addFailure("...",
                     "Namespace must ..")
  .withConfigProperty("namespace")
  .withConfigProperty("autoCreateTopic");
```

File: core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java [hunk 2]
```diff
@@ -136,13 +144,15 @@ private Conf() {
...
+                .withConfigProperty("autoCreateTopic");
       }
       if (maxAlertsPerSecond < 1) {
-        throw new IllegalArgumentException(
-          String.format("Invalid maxAlertsPerSecond %d. Must be at least 1.", maxAlertsPerSecond));
+        collector.addFailure(String.format("Invalid maxAlertsPerSecond %d. Must be at least 1.", maxAlertsPerSecond),
+                "").withConfigProperty("maxAlertsPerSecond");
```
Comment (albertshau) [inline core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java]: align "" with argument from previous line

**Review by albertshau — COMMENTED** [review]
looks like there was an unintentional change to the google-cloud submodule hash

File: core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java (hunk 1 above)
Comment (MEseifan) [reply core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java]: Done

File: core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java (hunk 2 above)
Comment (MEseifan) [reply core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java]: Done

File: core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java [hunk 3]
```diff
@@ -136,13 +144,15 @@ private Conf() {
       maxAlertsPerSecond = 100;
     }
 
-    private void validate() {
+    private void validate(FailureCollector collector) {
       if (autoCreateTopic && namespace != null) {
-        throw new IllegalArgumentException("Cannot auto-create topic when namespace is set.");
+        collector.addFailure("Cannot auto-create topic when namespace is set.",
+                             "Namespace must not be set to auto-create topic.").withConfigProperty("namespace")
+                .withConfigProperty("autoCreateTopic");
```
Comment (albertshau) [inline core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java]: method calls should be indented 2 spaces.

```
collector.addFailure("...",
                     "...")
  .withConfigProperty("namespace")
  .withConfigProperty("autoCreateTopic")
```

File: core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java [hunk 4]
```diff
@@ -136,13 +144,15 @@ private Conf() {
...
       }
       if (maxAlertsPerSecond < 1) {
-        throw new IllegalArgumentException(
-          String.format("Invalid maxAlertsPerSecond %d. Must be at least 1.", maxAlertsPerSecond));
+        collector.addFailure(String.format("Invalid maxAlertsPerSecond %d. Must be at least 1.", maxAlertsPerSecond),
+                            "").withConfigProperty("maxAlertsPerSecond");
```
Comment (albertshau) [inline core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java]: indentation (off by 1 space)

**Review by albertshau — COMMENTED** [review]
fix the google-cloud submodule hash

File: core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java (hunk 4 above)
Comment (MEseifan) [reply core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java]: Done

File: core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java (hunk 3 above)
Comment (MEseifan) [reply core-plugins/src/main/java/io/cdap/plugin/alert/TMSAlertPublisher.java]: Done

**Review by albertshau — APPROVED** [review]

**Comment by albertshau** [comment]
squashed commits, merging

**Pull request merged**